xarray/plot/dataset_plot.py

```diff
@@ -291,7 +291,7 @@ def newplotfunc(
             allargs = locals().copy()
             allargs["plotfunc"] = globals()[plotfunc.__name__]
             allargs["data"] = ds
-            # TODO dcherian: why do I need to remove kwargs?
+            # remove kwargs to avoid passing the information twice
```

Comment:
@dcherian, what should we do about this?

Reply:
Not sure :) the design is very confusing to me :)